### PR TITLE
Add testing for excessive SCC accesses on server

### PR DESCRIPTION
**Pull request by mtravitzky**
## What does this PR change?

This adds new tests for excessive SCC accesses. To enable, set `scc_access_logging = true` for the server.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes
- 
Maybe document this in internal readme documents?

- [x] **DONE**

## Test coverage
- Cucumber tests were added

- [x] **DONE**

## Links

Sumaform changes: https://github.com/uyuni-project/sumaform/pull/1519
Issue(s): https://github.com/SUSE/spacewalk/issues/22943
Port(s): https://github.com/SUSE/spacewalk/pull/23861

- [x] **DONE**

## Changelogs

- [x] No changelog needed

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: testsuite/features/finishing/srv_count_scc_access.feature [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# Licensed under the terms of the MIT license.
+
+@srv_scc_access_logging
+Feature: Test for excessive SCC accesses
+
+  Scenario: Count UI SCC accesses in rhn_web_ui.log
+    Then the word "scc.suse.com" does not occur more than 100 times in "/var/log/rhn/rhn_web_ui.log" on "server"
+
+  Scenario: Count Taskomatic SCC accesses in rhn_taskomatic_daemon.log
+    Then the word "scc.suse.com" does not occur more than 50 times in "/var/log/rhn/rhn_taskomatic_daemon.log" on "server"
```
Comment (nodeg) [inline testsuite/features/finishing/srv_count_scc_access.feature:11]:
```suggestion
    Then the word "scc.suse.com" does not occur more than 50 times in "/var/log/rhn/rhn_taskomatic_daemon.log" on "server"

```
A newline at the end is missing. The rest is simply a port from 4.3 I guess?

Comment (mtravitzky) [reply testsuite/features/finishing/srv_count_scc_access.feature]: Yup, just a direct port. The only thing that _might_ require changing are the threshold values, but sadly I don't have enough time left in this team, as today is very definitely my last day in SUMA QE.

File: testsuite/features/support/env.rb [hunk 2]
```diff
@@ -587,6 +587,18 @@ def process_code_coverage
   skip_this_scenario unless $is_containerized_server
 end
 
+# only test for excessive SCC accesses if SCC access is being logged
+Before('@srv_scc_access_logging') do
+  skip_this_scenario unless scc_access_logging_grain?
```
Comment (Bischoff) [inline testsuite/features/support/env.rb:592]: I would have preferred a global variable rather than running this function each time

**Review by Bischoff — COMMENTED** [review]
see suggestion

**Review by nodeg — APPROVED** [review]
Approving, since it is a direct port from 4.3 that was already approved.

**Comment by nodeg** [comment]
Rebase done.

**Pull request merged**